model_api/python/model_api/visualizer/scene/scene.py

```diff
@@ -0,0 +1,72 @@
+"""Scene object."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+from PIL import Image
+
+from model_api.visualizer.primitive import Overlay, Primitive
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from model_api.visualizer.layout import Layout
+
+
+class Scene:
+    """Scene object.
+
+    Used by the visualizer to render.
+    """
+
+    def __init__(
+        self,
+        base: Image,
+        overlay: Overlay | list[Overlay] | np.ndarray | None = None,
+        layout: Layout | None = None,
+    ) -> None:
+        self.base = base
+        self.overlay = self._to_overlay(overlay)
+        self.layout = layout
+
+    def show(self) -> Image: ...
+
+    def save(self, path: Path) -> None: ...
```

Comment:
Is there any particular reason the method is not implemented?

Reply:
I plan to update this method in the follow up PRs. Right now I am slowly adding scaffolds. I also want to add a unit test for this.